lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1034,6 +1021,24 @@ class HSAQueue final : public KalmarQueue
         return count;
     }
 
+
+    int isEmpty() override {
+        int count = 0;
+        // Have to walk asyncOps since it can contain null pointers.
+        for (int i = 0; i < asyncOps.size(); ++i) {
+            if (asyncOps[i] != nullptr) {
+                auto asyncOp = asyncOps[i];
```

Comment:
Likewise this could be written as:

```cpp
return not std::any_of(asyncOps.begin(), asyncOps.end(), [](auto&& asyncOp)
{
    return asyncOp != nullptr and 
        hsa_signal_load_relaxed(*(static_cast <hsa_signal_t*> (asyncOp->getNativeHandle()))) != 0;
});
```